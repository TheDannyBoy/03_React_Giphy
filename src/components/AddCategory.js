import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = evt => {
    setInputValue(evt.target.value);
  };

  const handleSubmit = evt => {
    evt.preventDefault();
    setCategories(categories => ([ inputValue, ...categories ]));
    setInputValue('');
  };

  return (<form onSubmit={handleSubmit}>
    <h1>{ inputValue }</h1>
    <input
      type='text'
      value={inputValue}
      onChange={handleInputChange}>
    </input>
  </form>)
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
