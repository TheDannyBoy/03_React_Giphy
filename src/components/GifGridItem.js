import React from 'react'

export const GifGridItem = ({ title, url }) => {
  const classes = ['card'];
  return (
    <div className={classes.join(' ')}>
      <img src={url} alt={title} />
      <p>{ title }</p>
    </div>
  )
}
