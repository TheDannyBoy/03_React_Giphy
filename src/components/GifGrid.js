import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import getGifs from '../helpers/getGifs'

export const GifGrid = ({ category }) => {
  useEffect(() => {
    getGifs(category).then(setImagesList);
  }, [category]);

  const [imagesList, setImagesList] = useState([]);

  return (
    <>
      <h3>{ category }</h3>
      <div className='card-grid'>
        {
          imagesList.map(image => (
            <GifGridItem key={image.id} {...image} />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string,
}
