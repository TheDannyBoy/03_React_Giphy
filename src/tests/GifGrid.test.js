import { shallow } from 'enzyme';
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs');

describe('Testing GifGrid', () => {
  const category = 'PokÃ©mon';

  test('Should create GifGrid correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot();
  })

  test('Should show items when images finished loading', () => {
    const gifs = [{
      id: 'foo',
      url: 'https://foo',
      title: 'bar',
    }, {
      id: 'bar',
      url: 'https://foo',
      title: 'bar',
    }];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  })
  
});
