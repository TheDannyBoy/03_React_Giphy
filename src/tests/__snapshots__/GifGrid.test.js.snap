// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid Should create GifGrid correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    Pokémon
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing GifGrid Should show items when images finished loading 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    Pokémon
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="foo"
      key="foo"
      title="bar"
      url="https://foo"
    />
    <GifGridItem
      id="bar"
      key="bar"
      title="bar"
      url="https://foo"
    />
  </div>
</Fragment>
`;
